Sources/NIOHTTPTypesHTTP1/HTTPTypeConversion.swift

```diff
@@ -0,0 +1,207 @@
+            case .PROPFIND: return HTTPRequest.Method("PROPFIND")!
+            case .CHECKOUT: return HTTPRequest.Method("CHECKOUT")!
+            case .PROPPATCH: return HTTPRequest.Method("PROPPATCH")!
+            case .SUBSCRIBE: return HTTPRequest.Method("SUBSCRIBE")!
+            case .MKCALENDAR: return HTTPRequest.Method("MKCALENDAR")!
+            case .MKACTIVITY: return HTTPRequest.Method("MKACTIVITY")!
+            case .UNSUBSCRIBE: return HTTPRequest.Method("UNSUBSCRIBE")!
+            case .SOURCE: return HTTPRequest.Method("SOURCE")!
+            case .RAW(value: let value):
+                guard let method = HTTPRequest.Method(value) else {
+                    throw HTTP1TypeConversionError.invalidMethod
+                }
+                return method
+            }
+        }
+    }
+}
+
+extension HTTPHeaders {
+    init(_ newFields: HTTPFields) {
+        let fields = newFields.map { ($0.name.rawName, $0.value) }
+        self.init(fields)
+    }
+
+    func newFields(splitCookie: Bool) -> HTTPFields {
+        var fields = HTTPFields()
+        fields.reserveCapacity(count)
+        for (index, field) in enumerated() {
+            if index == 0, field.name.lowercased() == "host" {
+                continue
+            }
+            if let name = HTTPField.Name(field.name) {
+                if splitCookie && name == .cookie, #available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, *) {
+                    fields.append(contentsOf: field.value.split(separator: "; ", omittingEmptySubsequences: false).map {
+                        HTTPField(name: name, value: String($0))
+                    })
+                } else {
+                    fields.append(HTTPField(name: name, value: field.value))
+                }
+            }
+        }
+        return fields
+    }
+}
+
+extension HTTPRequestHead {
+    init(_ newRequest: HTTPRequest) throws {
+        guard let path = newRequest.method == .connect ? newRequest.authority : newRequest.path else {
+            throw HTTP1TypeConversionError.missingPath
+        }
+        var headers = HTTPHeaders()
+        headers.reserveCapacity(newRequest.headerFields.count + 1)
+        if let authority = newRequest.authority {
+            headers.add(name: "Host", value: authority)
+        }
+        var firstCookie = true
+        for field in newRequest.headerFields {
+            if field.name == .cookie {
+                if firstCookie {
+                    firstCookie = false
+                    headers.add(name: field.name.rawName, value: newRequest.headerFields[.cookie]!)
+                }
+            } else {
+                headers.add(name: field.name.rawName, value: field.value)
+            }
+        }
+        self.init(
+            version: .http1_1,
+            method: HTTPMethod(newRequest.method),
+            uri: path,
+            headers: headers
+        )
+    }
+
+    func newRequest(secure: Bool, splitCookie: Bool) throws -> HTTPRequest {
```

Comment:
We normally spell this as an `init` on `HTTPRequest` that takes an `HTTPRequestHead` and the two other parameters.